### PR TITLE
Add Decimal Type documentation

### PR DESCRIPTION
**Pull request by majetideepak**
Document Decimal type and its function signature.
Create a separate page for Types.

### CODE REVIEW

File: velox/docs/develop/simple-vector-functions.rst [hunk 1]
```diff
@@ -1,6 +1,6 @@
-=============================
-How to add a scalar function?
-=============================
+=======================================
+How to add a simple or vector function?
```
Comment (mbasmanova) [inline velox/docs/develop/simple-vector-functions.rst:2]: I think "scalar" is a better term to use in the title.

File: velox/docs/develop/types.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
+=====
+Types
+=====
+
+Velox supports scalar types and complex types. These types cover both
```
Comment (mbasmanova) [inline velox/docs/develop/types.rst:5]: > These types cover both Presto and Spark data types.

This sentence doesn't seem accurate. Velox types cover only a subset of Presto and Spark types. There are also application-specific types like TIMESTAMP WITH TIME ZONE.

Comment (mbasmanova) [inline velox/docs/develop/types.rst:5]: Would it make sense to explain the difference between scalar and complex types? E.g. complex types contain other types.

File: velox/docs/develop/types.rst [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+All the three values, precision, scale, unscaled value are required to represent a
+decimal value. UnscaledLongDecimal is a wrapper around the `int128_t` type.
+Some systems implement a similar wrapper around two `int64_t` values.
+Velox chose `int128_t` since most compilers now support this type and
+it simplifies the implementation.
+
+
+Complex Types
+~~~~~~~~~~~~~
+Velox supports the ARRAY, MAP, and ROW / STRUCT complex types.
```
Comment (mbasmanova) [inline velox/docs/develop/types.rst:54]: Perhaps, remove / STRUCT. We can explain that ROW type is used for structs separately.

File: velox/docs/develop/types.rst [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+Their corresponding C++ types, UnscaledShortDecimal and UnscaledLongDecimal carry
+the unscaled value. For example, the unscaled value of decimal `123.45` is `12345`.
+All the three values, precision, scale, unscaled value are required to represent a
+decimal value. UnscaledLongDecimal is a wrapper around the `int128_t` type.
+Some systems implement a similar wrapper around two `int64_t` values.
+Velox chose `int128_t` since most compilers now support this type and
+it simplifies the implementation.
+
+
+Complex Types
```
Comment (mbasmanova) [inline velox/docs/develop/types.rst:52]: There are also UNKNOWN, OPAQUE and FIXED_SIZE_ARRAY types.

**Review by mbasmanova — COMMENTED** [review]
@majetideepak Thank you for adding Types article. It was long overdue.

File: velox/docs/develop/simple-vector-functions.rst (hunk 1 above)
Comment (majetideepak) [reply velox/docs/develop/simple-vector-functions.rst:2]: To clarify, keep the previous `How to add a scalar function?` or make this `How to add a scalar or vector function?`

File: velox/docs/develop/types.rst (hunk 2 above)
Comment (majetideepak) [reply velox/docs/develop/types.rst:5]: I will update to `These types cover most Presto and Spark data types`.

Comment (majetideepak) [reply velox/docs/develop/types.rst:5]: Complex Types definitely need more love. I plan to focus on Complex Types with examples in a follow-up PR.
Is that okay?

File: velox/docs/develop/types.rst (hunk 3 above)
Comment (majetideepak) [reply velox/docs/develop/types.rst:54]: Should we remove ROW and keep STRUCT instead?

Comment (majetideepak) [reply velox/docs/develop/types.rst:54]: Never mind. I see that ROW is commonly used.

File: velox/docs/develop/simple-vector-functions.rst (hunk 1 above)
Comment (mbasmanova) [reply velox/docs/develop/simple-vector-functions.rst]: I'd keep the original "scalar function".

**Review by mbasmanova — APPROVED** [review]

**Comment by mbasmanova** [comment]
@majetideepak For some reason CI jobs are red.

**Comment by majetideepak** [comment]
@mbasmanova seems to be a CI glitch. I amended my last commit and pushed again.

**Pull request closed without merging**